zfs.go

```diff
@@ -102,10 +103,16 @@ func cloneFilesystem(datasetName string, snapshot *zfs.Dataset) (*zfs.Dataset, e
 }
 
 func destroy(dataset *zfs.Dataset) error {
+	if strings.Contains(dataset.Name, "@") {
```

Comment:
Is it possible to have a dataset with an escaped `@`?

Reply:
removed this part. can be another PR.